src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -0,0 +1,152 @@
+    /// to describe their type, which is expected to be provided by development tools. For names of members,  
+    /// if a data type name must be used, use a language-independent name instead of a language-specific one.  
+    /// </summary> 
+    
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class IdentifiersShouldNotContainTypeNames : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1720";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.IdentifiersShouldNotContainTypeNamesTitle), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.IdentifiersShouldNotContainTypeNamesMessage), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.IdentifiersShouldNotContainTypeNamesDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static HashSet<string> s_types = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Design,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        /// <summary> 
+        /// Initializes primitive types 
+        /// </summary> 
+        static IdentifiersShouldNotContainTypeNames()
+        {
+            s_types.Add("bool");
+            s_types.Add("boolean");
+            s_types.Add("byte");
+            s_types.Add("sbyte");
+            s_types.Add("ubyte");
```

Comment:
Inline the static field initialization might provide better performance. See https://msdn.microsoft.com/en-us/library/ms182275.aspx